### PR TITLE
Broad schema updates and ASv4

### PR DESCRIPTION
**Pull request by lennyburdette**
* Update with some best practices for schema design
* Use Apollo Server v4

### CODE REVIEW

**Review by kschrade — COMMENTED** [review]
Should we be using RCs or just wait for them to get to GA?

File: gateway/server.js [hunk 1]
```diff
@@ -14,17 +14,17 @@ export const start = async (port) => {
...
-    plugins: [ApolloServerPluginLandingPageLocalDefault({ embed: true })],
   });
 
   const serverPort = port ?? process.env.PORT;
-  const { url } = await server.listen(serverPort);
+  const { url } = await startStandaloneServer(server, {
+    listen: { port: serverPort },
+    async context({ req }) {
+      return { headers: req.headers };
+    },
```
[the remark this answers is not in the document]
Comment (lennyburdette) [reply gateway/server.js:26]: what do you mean by "non-named parameter"?

**Comment by lennyburdette** [comment]
> Should we be using RCs or just wait for them to get to GA?

I'm comfortable using the RC because these basic APIs aren't going to change.

File: gateway/server.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (kschrade) [reply gateway/server.js:26]: this is a positional argument (2nd parameter in the function call) vs. what it was before where you have a named context key you are making this equal to like the following:

```Javascript
const server = new ApolloServer({
  ...
  context: async (thing) => functionToMakeContext(thing),
  ...
}); 
```

[the remark this answers is not in the document]
Comment (kschrade) [reply gateway/server.js:26]: Wait never mind - it is just put in a little weird.

**Review by smyrick — APPROVED** [review]

File: gateway/server.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (lennyburdette) [reply gateway/server.js:26]: ah yeah, i prefer to use `{ foo() { ... } }` instead of `{ foo: () => {} }` unless i _need_ to close over `this`

**Comment by smyrick** [comment]
For reference, we missed some step in the migration guide: https://github.com/apollosolutions/retail-supergraph/pull/63

**Pull request merged**